src/App.tsx

```diff
@@ -64,15 +91,14 @@ function App() {
         placeholder="Enter github username"
         className="input-field"
         value={userName}
-        onChange={(e) => setUserName(e.target.value.trim())}
-        onKeyUp={() => handleKeyUp()}     
+        onChange={(e) => handleOnChange(e)}     
       />
+      {userData && error && <p>User {error}</p>}
+
+      {!error && userData.login && <UserProfile
```

Comment:
Can we initialise userData with null value & check on that only, instead of checking for error & userData both

Reply:
Instead of checking for both, i am just for checking for error now.
How can we set userData to null? because the type of userData is User (which is the interface of data we get from API) and we have kept some of the data as mandatory. Also, i have already initialized the required properties as null.